sdk/src/azure/platform/az_mosquitto5.c

```diff
@@ -340,7 +340,7 @@ AZ_NODISCARD az_result az_mqtt5_outbound_sub(az_mqtt5* mqtt5, az_mqtt5_sub_data*
       (char*)az_span_ptr(sub_data->topic_filter),
       sub_data->qos,
       0,
-      sub_data->properties ? sub_data->properties->properties : NULL));
+      (sub_data->properties == NULL) ? NULL : sub_data->properties->properties));
```

Comment:
We are currently calling `mosquitto_connect_async` to connect which does not accept properties. From what I checked, there is not a `mosquitto_connect_async_v5` in the mosquitto API. The closest would be `mosquitto_connect_bind_v5` which takes different parameters. Since at the moment we are not using properties on connect I am keeping it as is.